### ISSUE TITLE
[Variant] Ko Shogi

### ISSUE THREAD

**Issue by Eventlesstew** [WandererXII/lishogi#828]
![Ko Shogi](https://github.com/WandererXII/lishogi/assets/161836208/f4cea017-0ef0-4dad-84ea-6940d12d31dd)
(My Ko Shogi set.)

SOURCE: https://en.wikipedia.org/wiki/Ko_shogi

Ko Shogi is a 19x19 variant of Shogi that is completely different. Pieces move on vertices similar to Xiangqi and there are more complex rules on capturing and promoting.

Certain pieces promote upon capturing other certain pieces.
Some pieces can shoot.
Some pieces can't capture the Heavenly Fortress via displacement.

Basically, it's a very complex variant not just in board size and piece quantity, but rules-wise, so learning & playing it is not for the faint of heart.

**Comment by Eventlesstew** [comment]
I can defs see this getting closed in a heartbeat lol

**Comment by WandererXII** [comment]
Just reading the rules... This would be crazy to implement lol - shooting, burning, quintuple moves, piece behaving differently based on pieces near them... Nope:D

**Comment by Eventlesstew** [comment]
lol
